pkg/cloudprovider/suite_test.go

```diff
@@ -233,5 +238,62 @@ var _ = Describe("CloudProvider", func() {
 			Expect(err).ToNot(HaveOccurred())
 			Expect(string(drifted)).To(Equal("ImageVersionDrift"))
 		})
+
+		Context("Kubernetes Version", func() {
+			It("should succeed with no drift when nothing changes", func() {
+				drifted, err := cloudProvider.IsDrifted(ctx, nodeClaim)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(drifted).To(Equal(NoDrift))
+			})
+
+			It("should succeed with no drift when KubernetesVersionReady is not true", func() {
+				nodeClass = ExpectExists(ctx, env.Client, nodeClass)
+				nodeClass.StatusConditions().SetFalse(v1alpha2.ConditionTypeKubernetesVersionReady, "K8sVersionNoLongerReady", "test when k8s isn't ready")
+				ExpectApplied(ctx, env.Client, nodeClass)
+				drifted, err := cloudProvider.IsDrifted(ctx, nodeClaim)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(drifted).To(Equal(NoDrift))
+			})
+
+			// TODO (charliedmcb): I'm wondering if we actually want to have these soft-error cases switch to return an error if no-drift condition was found.
+			It("shouldn't error or be drifted when KubernetesVersion is empty", func() {
+				nodeClass = ExpectExists(ctx, env.Client, nodeClass)
+				nodeClass.Status.KubernetesVersion = ""
+				ExpectApplied(ctx, env.Client, nodeClass)
+				drifted, err := cloudProvider.IsDrifted(ctx, nodeClaim)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(drifted).To(Equal(NoDrift))
+			})
+
+			// TODO (charliedmcb): I'm wondering if we actually want to have these soft-error cases switch to return an error if no-drift condition was found.
+			It("shouldn't error or be drifted when NodeName is missing", func() {
+				nodeClaim.Status.NodeName = ""
+				drifted, err := cloudProvider.IsDrifted(ctx, nodeClaim)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(drifted).To(Equal(NoDrift))
+			})
+
+			// TODO (charliedmcb): wondering if we should actually ignore not found errors? Can IsDrifted be called before the node exists?
```

Comment:
```
	// From here there are three scenarios to handle:
	// 1. If NodeClaim is not launched, remove the drift status condition
	if !nodeClaim.StatusConditions().Get(v1.ConditionTypeLaunched).IsTrue() {
		_ = nodeClaim.StatusConditions().Clear(v1.ConditionTypeDrifted)
		if hasDriftedCondition {
			log.FromContext(ctx).V(1).Info("removing drift status condition, isn't launched")
		}
		return reconcile.Result{}, nil
	}
```

Suggests no

Reply:
(from: pkg/controllers/nodeclaim/disruption/drift.go)

Reply:
Actually that makes me think that we might be able to reach this point while the node doesn't exist, as the cases you pointed to us clearing the Drifted condition meaning that it would have had to get there while the it was not launched

Reply:
Oh, no you are right. The call for IsDrifted happens after the fix block you showed:
https://github.com/kubernetes-sigs/karpenter/blob/9877cf639e665eadcae9e46e5a702a1b30ced1d3/pkg/controllers/nodeclaim/disruption/drift.go#L58

Should be fine than.

Reply:
Removed TODO, and left a comment answering the question at the actual code location NotFound would occur.

Reply:
So, I actually do think we can get here when the node doesn't exist yet.

In core, the Launched status will become ready after we get a successful return from the cloudprovider Create:
https://github.com/kubernetes-sigs/karpenter/blob/9877cf639e665eadcae9e46e5a702a1b30ced1d3/pkg/controllers/nodeclaim/lifecycle/launch.go#L71

Right now, this won't really be an issue we encounter, but won't we fix the LRO issue, I believe that will mean we can get here before registration happens.

Reply:
Deferred as following up: Gracefully handle `NodeNodeFound` and `DeletionTimestamp` in K8s Drift